tests/test_query.py

```diff
@@ -115,3 +117,16 @@ def test_get_latest_versions(
     result = get_latest_versions(url, ids)
 
     assert result == expected
+
+
+@pytest.mark.parametrize(
+    "ids, latest_ids", [(["foo", "bar"], ["foo", "baz"])],
+)
+def test_get_latest_versions_error(
+    versions_response_error, ids: List[str], latest_ids: List[str],
+) -> None:
+    url = "https://example.com"
+    versions_response_error(url + "/files/versions")
+
+    with pytest.raises(Exception):
```

Comment:
Shouldn't we make sure that it's throwing the expected JSONDecodeError? Also this might be a good time to use Mocks to insure that the logger.error was at least called too.